core/module.h

```diff
@@ -18,6 +19,14 @@ using bess::gate_idx_t;
 #define MAX_NUMA_NODE 16
 #define UNCONSTRAINED_SOCKET ((0x1ull << MAX_NUMA_NODE) - 1)
 
+class Node {
+ public:
+  Node(Module *module) : module_(module), children_(), parents_() {}
```

Comment:
Add comments for each method (including the constructor and its args) and for each member field.